src/ManagedServices/ManagedServices/Commands/NewAzureRmManagedServicesAssignment.cs

```diff
@@ -31,24 +31,20 @@ namespace Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Commands
     public class NewAzureRmManagedServicesAssignment : ManagedServicesCmdletBase
     {
         protected const string DefaultParameterSet = "Default";
-        protected const string ByResourceIdParameterSet = "ByResourceId";
         protected const string ByInputObjectParameterSet = "ByInputObject";
 
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The unique name of the Registration Assignment.")]
+        public string Name { get; set; }
+
         [Parameter(Position = 0, ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The scope where the registration assignment should be created.")]
-        [Parameter(Position = 0, ParameterSetName = ByResourceIdParameterSet, Mandatory = false, HelpMessage = "The scope where the registration assignment should be created.")]
         [Parameter(Position = 0, ParameterSetName = ByInputObjectParameterSet, Mandatory = false, HelpMessage = "The scope where the registration assignment should be created.")]
         [ScopeCompleter]
         public string Scope { get; set; }
 
-        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The registration definition identifier.")]
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The unique name of the Registration Definition.")]
         [ValidateNotNullOrEmpty]
         public string RegistrationDefinitionName { get; set; }
 
-        [Parameter(ParameterSetName = ByResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "The fully qualified resource id of the registration definition.")]
-        [ValidateNotNullOrEmpty]
-        [Alias("ResourceId")]
-        public string RegistrationDefinitionResourceId { get; set; }
```

Comment:
removing mandatory parameter is breaking change and is not allowed in non-breaking change release